extensions/amp-form/0.1/amp-form.js

```diff
@@ -440,17 +440,20 @@ export class AmpForm {
    */
   doXhr_(opt_extraValues) {
     const isHeadOrGet = this.method_ == 'GET' || this.method_ == 'HEAD';
+    const values = this.getFormAsObject_(opt_extraValues);
+    this.renderTemplate_(values);
```

Comment:
Justin's assessment is correct, it only render's the current state's template. In this case it's the `FormState_.SUBMITTING` state.